python-checks/src/main/java/org/sonar/python/checks/ItemOperationsTypeCheck.java

```diff
@@ -70,6 +69,48 @@ public boolean isValidSubscription(Expression subscriptionObject, String require
     return type.canHaveMember(requiredMethod);
   }
 
+  @CheckForNull
+  private static Boolean isValidSubscriptionSymbol(HasSymbol hasSymbol, Expression subscriptionObject, Map<LocationInFile, String> secondaries, String requiredMethod,
+    @Nullable String classRequiredMethod) {
+    Symbol symbol = hasSymbol.symbol();
+    if (symbol == null || isTypingOrCollectionsSymbol(symbol)) {
+      return true;
+    }
+    if (symbol.is(FUNCTION, CLASS)) {
+      secondaries.put(symbol.is(FUNCTION) ? ((FunctionSymbol) symbol).definitionLocation() : ((ClassSymbol) symbol).definitionLocation(),
+        String.format(SECONDARY_MESSAGE, symbol.name()));
+
+      if (isSubscriptionInClassArg(subscriptionObject).isPresent()) {
+        return true;
+      }
+
+      return canHaveMethod(symbol, requiredMethod, classRequiredMethod);
+    }
+    return null;
+  }
+
+  private static boolean isValidSubscriptionCallExpr(CallExpression subscriptionObject, Map<LocationInFile, String> secondaries) {
```

Comment:
could be simplified
```java
private static boolean isValidSubscriptionCallExpr(Expression expression, Map<LocationInFile, String> secondaries) {
    if (expression instanceof CallExpression callExpression
        && callExpression.calleeSymbol() instanceof FunctionSymbol functionSymbol
        && functionSymbol.isAsynchronous()) {
      secondaries.put(functionSymbol.definitionLocation(), SECONDARY_MESSAGE.formatted(functionSymbol.name()));
      return true;
    }
    return false;
  }

```
and then no need to check for `subscriptionObject.is(Tree.Kind.CALL_EXPR)` on line 56